Sockets/BackEndMonitoringSockets/include/CAcceptor.h

```diff
@@ -1,23 +1,29 @@
 #pragma once
-#include "CAcceptorSocket.h"
-#include "CServiceConnectionHandler.h"
-#include "CServiceHandler.h"
+#include "CSocket.h"
+
+class CServiceConnectionHandler;
+class CServiceHandler;
+
 // Class for accepting connection by the server
 class CAcceptor
 {
 public:
-	CAcceptor(const int port, const std::string& ip_address, 
-		std::shared_ptr<CLogger> logger);
-	int GetConnectedHandle();
+	CAcceptor(const int port, const std::string& ip_address, bool is_blocked);
```

Comment:
CAcceptor class should have next public methods:
CAcceptor(const SAcceptorParams &); // timeout and is_blocked
bool Initialize(const std::string& ip_address, const int listener_port, const int connections); //last parameter for listen function
bool Accept(CSocketWrapper *& p_connection_socket);
bool IsTimeOutWithoutConnections(); // returns true if ::select returns timeout needed for AcceptorWrapper 

All other functions should be private or protected.